supervisor.c

```diff
@@ -158,6 +158,19 @@ int get_digits_num(const int number)
    return cnt;
 }
 
+// Checks if str ends with suffix.
+int strsuffixis(const char *str, const char *suffix)
```

Comment:
Your function does not take in account the option "file.sup.sup".
I think, the whole function could look like this:

`int strsuffixis(const char *str, char *suffix) {`
`   return strcmp(str + strlen(str) - strlen(suffix), suffix) == 0;`
`}`

Reply:
Agreed! Please see the updated commit.